crates/rust-eigenda-v2-client/src/core/blob_key.rs

```
@@ -41,7 +40,7 @@ impl BlobKey {
                 "Invalid hex string length".to_string(),
             ));
         }
-        Ok(BlobKey(bytes.try_into().unwrap()))
+        Ok(BlobKey::from_bytes(bytes.try_into().unwrap())) // Safe unwrap as we checked the length
```

you can return an error as well

---

perfect